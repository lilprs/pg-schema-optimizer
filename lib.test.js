import { test, expect } from 'bun:test'
import { optimize_schema } from './lib'

test('optimize_schema', () => {
  expect(
    optimize_schema(`
  CREATE TABLE public.test (
    "id" int NOT NULL,
    "varchar" varchar(255) NOT NULL,
    "text" text NOT NULL,
    "boolean" boolean NOT NULL,
    "nullable_boolean" boolean,
    "date" date NOT NULL,
    "timestamp" timestamp NOT NULL,
    "timestamptz" timestamptz NOT NULL,
    "nullable_timestamptz" timestamptz,
    "nullable_timestamptz_with_default" timestamptz DEFAULT now(),
    "time" time NOT NULL,
    "timetz" timetz NOT NULL,
    "interval" interval NOT NULL,
    "numeric" numeric NOT NULL,
    "decimal" decimal NOT NULL,
    "real" real NOT NULL,
    "double" double precision NOT NULL,
    "money" money NOT NULL,
    "smallint" smallint NOT NULL,
    "integer" integer NOT NULL,
    "nullable_integer" integer,
    "bigint" bigint NOT NULL,
    "serial" serial NOT NULL,
    "bigserial" bigserial NOT NULL,
    "bit" bit NOT NULL,
    "bit_varying" bit varying NOT NULL,
    "uuid" uuid NOT NULL,
    "xml" xml NOT NULL,
    "json" json NOT NULL,
    "jsonb" jsonb NOT NULL,
    "point" point NOT NULL,
    "line" line NOT NULL,
    "lseg" lseg NOT NULL,
    "box" box NOT NULL,
    "path" path NOT NULL,
    "polygon" polygon NOT NULL,
    "circle" circle NOT NULL,
    "cidr" cidr NOT NULL,
    "inet" inet NOT NULL
    );
  `),
  ).toMatchSnapshot()
})
