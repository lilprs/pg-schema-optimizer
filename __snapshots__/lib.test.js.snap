// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`optimize_schema 1`] = `
"
  CREATE TABLE public.test (
    "bigint" bigint NOT NULL,
    "bigserial" bigserial NOT NULL,
    "box" box NOT NULL,
    "circle" circle NOT NULL,
    "double" double precision NOT NULL,
    "interval" interval NOT NULL,
    "line" line NOT NULL,
    "lseg" lseg NOT NULL,
    "money" money NOT NULL,
    "path" path NOT NULL,
    "point" point NOT NULL,
    "polygon" polygon NOT NULL,
    "time" time NOT NULL,
    "timestamp" timestamp NOT NULL,
    "timestamptz" timestamptz NOT NULL,
    "timetz" timetz NOT NULL,
    "uuid" uuid NOT NULL,
    "nullable_timestamptz_with_default" timestamptz DEFAULT now(),
    "nullable_timestamptz" timestamptz,
    "cidr" cidr NOT NULL,
    "date" date NOT NULL,
    "id" int NOT NULL,
    "inet" inet NOT NULL,
    "integer" integer NOT NULL,
    "real" real NOT NULL,
    "serial" serial NOT NULL,
    "nullable_integer" integer,
    "smallint" smallint NOT NULL,
    "bit" bit NOT NULL,
    "boolean" boolean NOT NULL,
    "nullable_boolean" boolean,
    "bit_varying" bit varying NOT NULL,
    "decimal" decimal NOT NULL,
    "json" json NOT NULL,
    "jsonb" jsonb NOT NULL,
    "numeric" numeric NOT NULL,
    "text" text NOT NULL,
    "varchar" varchar(255) NOT NULL,
    "xml" xml NOT NULL
    );
  
"
`;

exports[`optimize_schema 2`] = `
"
    CREATE TABLE "test" (
      last_price_update timestamp with time zone DEFAULT now() NOT NULL,
      last_quantity_update timestamp with time zone DEFAULT now() NOT NULL,
      "productId" integer DEFAULT 0 NOT NULL,
      price integer NOT NULL
    )
    WITH (autovacuum_vacuum_scale_factor='0.0', autovacuum_vacuum_threshold='100', autovacuum_analyze_scale_factor='0.0', autovacuum_analyze_threshold='100');
  
"
`;
